### PR TITLE
simplifications to the typing of basefluent

### PR DESCRIPTION
**Pull request by shawkins**
fixes fabric8io/kubernetes-client#5522

### CODE REVIEW

**Comment by shawkins** [comment]
@iocanel the core of the issue is that implementing Visitable twice is confusing the compiler, this removes that but also does a few other things:
- removes the Fluent interface, we don't seem to need it any more
- simplifies BaseFluent to just BaseFluent<F> - we don't really need a bound on this because we only ever use it via SomethingFluent, we could even make it abstract.
- adds a getVisitableMap() on Visitables so that we can pushdown all of the visitable logic.  However I took a shortcut and put the implementation of this method on BaseFluent.

Also I'm assuming that SomethingFluent does not actually need to be Visitable, but I could be wrong about that.

**Comment by shawkins** [comment]
@iocanel are the any changes that you want on this pr?  We'd like to get it into a fabric8 release so that the operator sdk can upgrade without changing their code.

File: annotations/builder/src/main/java/io/sundr/builder/internal/BuilderContext.java [hunk 1]
```diff
@@ -1077,66 +1074,8 @@ public BuilderContext(Elements elements, Types types, Boolean generateBuilderPac
 
         .addNewMethod()
         .withNewModifiers().withPublic().endModifiers()
-        .withName("accept")
```
Comment (iocanel) [inline annotations/builder/src/main/java/io/sundr/builder/internal/BuilderContext.java:1080]: While we do drop this large chunk of boilerplate and I am grateful for it, I cannot help wonder what exactly allow us to do so.

I understand that we moved the target methods from BaseFluent to Visitable, but don't we need to move their definition too? In other words  should we construct a Visitable TypeDef that includes these `accept` methods?

**Comment by iocanel** [comment]
@shawkins I did add a question.

File: annotations/builder/src/main/java/io/sundr/builder/internal/BuilderContext.java (hunk 1 above)
Comment (shawkins) [reply annotations/builder/src/main/java/io/sundr/builder/internal/BuilderContext.java:1080]: The Visitable TypeDef already has the necessary logic to support the accept default methods.  The only new method is getVisitableMap, which handling was added for on both Visitable and BaseFluent.  The one bit of messiness there is that it has to be added to Visitable after VisitableMap is defined - is there a better way to do that?

**Comment by iocanel** [comment]
BTW @shawkins: excellent job. I salute!

**Comment by shawkins** [comment]
> BTW @shawkins: excellent job. I salute!

Thank you @iocanel - it was good to learn more of the code, you have a humbling amount of work here.

**Pull request merged**